api/serializers/site.js

```diff
@@ -1,26 +1,30 @@
-const { Site } = require("../models")
+const { Site, User } = require('../models');
 
-const serialize = (serializable) => {
-  if (serializable.length !== undefined) {
-    const siteIds = serializable.map(site => site.id)
-    const query = Site.findAll({ where: { id: siteIds } })
+const serializeObject = (site) => {
+  const json = site.toJSON();
+
+  if (json.Users) {
+    json.users = site.Users.map(u => u.toJSON());
+    delete json.Users;
+  }
 
-    return query.then(sites => {
-      return sites.map(site => serializeObject(site))
-    })
-  } else {
-    const site = serializable
-    const query = Site.findById(site.id)
+  return json;
+};
 
-    return query.then(site => {
-      return serializeObject(site)
-    })
+const serialize = (serializable) => {
+  // TODO: Why does this (and every other serializer) re-query for the model?
```

Comment:
@jmhooper: any insight into this question? It looks like all the serializers actually end up re-fetching models from the database when invoked. This seems a little odd since they are all called with database models already, so it seems like a wasted DB call(s).

Reply:
IIRC this was in here to maintain some Waterline behavior which would load associated objects when you called `req.json(model)`.

Reply:
hrm, ok, so I guess then it would be ok to just move that behavior into the controller methods as applicable.

Or maybe there is some way in Sequelize to tell it to always load some associations. I'll look into it.